### PR TITLE
Google APIs: fix link and add instructions for how to enable APIs

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: src/docs/development/data-and-backend/google-apis.md [hunk 1]
```diff
@@ -35,14 +35,16 @@ Examples of user-data APIs include
...
+1. Enable the API
 1. Authenticate user with the required scopes
 1. Obtain an authenticated HTTP client
 1. Create and use the desired API class
 
 ## 1. Pick the desired API
 
-The documentation for [package:googleapis] lists each API as a separate Dart
-library – in a `name.version` format. Let's look at
+The documentation for [package:googleapis](https://pub.dev/packages/googleapis)
```
Comment (kevmoo) [inline src/docs/development/data-and-backend/google-apis.md:45]: not sure how we missed this! 😟

**Review by domesticmouse — APPROVED** [review]

File: src/docs/development/data-and-backend/google-apis.md [hunk 2]
```diff
@@ -65,7 +66,19 @@ when authenticating the user.
...
+
+To use Google APIs you must have a Google account and a Google project. You also
+need to enable your desired API.
+
+In this example, you'd enable
+[YouTube Data API v3](https://console.cloud.google.com/apis/api/youtube.googleapis.com).
+
+Follow the
+[getting started instructions](https://cloud.google.com/apis/docs/getting-started)
+to learn more.
```
Comment (kwalrath) [inline src/docs/development/data-and-backend/google-apis.md:79]: Are we following instructions or just learning? Maybe:

```suggestion
For details, see the
[getting started instructions](https://cloud.google.com/apis/docs/getting-started).
```

File: src/docs/development/data-and-backend/google-apis.md [hunk 3]
```diff
@@ -133,3 +146,9 @@ var favorites = await youTubeApi.playlistItems.list(
   playlistId: 'LL', // Liked List
 );
 ```
+
+## More information
+
+- The
+  [`extension_google_sign_in_as_googleapis_auth` example](https://pub.dev/packages/extension_google_sign_in_as_googleapis_auth/example)
```
Comment (kwalrath) [inline src/docs/development/data-and-backend/google-apis.md:153]:
```suggestion
  [`extension_google_sign_in_as_googleapis_auth` example]({{site.pub-pkg}}/extension_google_sign_in_as_googleapis_auth/example)
```

File: src/docs/development/data-and-backend/google-apis.md (hunk 2 above)
Comment (kevmoo) [reply src/docs/development/data-and-backend/google-apis.md]: Following instructions. Just didn't want to duplicate a bunch of content that may evolve.

**Review by kwalrath — APPROVED** [review]

**Review by sfshaza2 — APPROVED** [review]
LGTM

Are you ready for me to merge this?

**Comment by kevmoo** [comment]
> LGTM
> 
> Are you ready for me to merge this?

Please!

**Pull request merged**